pkg/util/test/util.go

```diff
@@ -425,6 +425,11 @@ func ValidateStorageCluster(
 		return err
 	}
 
+	if _, ok := liveCluster.Annotations["portworx.io/migration-approved"]; ok {
+		logrus.Debug("Sleeping 10 seconds to ensure all components are bounced before validation")
+		time.Sleep(10 * time.Second)
```

Comment:
change to sleep 1s

Reply:
oops, i thought it has retry, we have too many sleep in code, the perf is very slow now. this should be a PollImmediate with timeout, even 10 seconds may not be enough occasionally. can you verify by retry with timeout, the retry interval should be 1 second or less.

Reply:
that makes the change complicated, this is just to make sure we don't get the old objects from daemonset, as after px pods are ready we'll create new component deployments, and the validations already have retries, also it's for migration tests only, IMO it should be good.

Reply:
it wont be a complicated change, just verify if ds is gone with pollImmediate. if the migration is not finished, the validation would still pass as the deployment and images are the same?

Reply:
sounds good, that case an easier way to do this is wait for ds is gone then validate the stc, we did this in a reverse way before. updated